components/tidb_query_expr/src/impl_math.rs

```diff
@@ -616,6 +616,9 @@ fn is_valid_base(base: IntWithSign) -> bool {
 
 fn extract_num_str(s: &str, from_base: IntWithSign) -> Option<(String, bool)> {
     let mut iter = s.chars().peekable();
+    if iter.peek().is_none() {
+        return None;
+    }
     let head = *iter.peek().unwrap();
```

Comment:
It can be simplify it as:
```suggestion
    let head = *iter.peek()?;
```